dbms/src/Functions/repeat.cpp

```diff
@@ -0,0 +1,236 @@
+            res_offsets.assign(offsets);
+            for (UInt64 i = 0; i < col_num.size(); ++i)
+			{
+                data_size += (offsets[i] - offsets[i - 1] - 1) * col_num[i] + 1;
+                res_offsets[i] = data_size;
+            }
+            res_data.resize(data_size);
+            for (UInt64 i = 0; i < col_num.size(); ++i)
+			{
+                array(data.data() + offsets[i - 1], res_data.data() + res_offsets[i - 1], offsets[i] - offsets[i - 1], col_num[i]);
+            }
+        }
+
+
+        template <typename T>
+        static void vectorNonConstInteger(
+            const String & copy_str, ColumnString::Chars & res_data, ColumnString::Offsets & res_offsets, const PaddedPODArray<T> & col_num)
+        {
+            UInt64 data_size = 0;
+            res_offsets.resize(col_num.size());
+            UInt64 str_size = copy_str.size();
+            for (UInt64 i = 0; i < col_num.size(); ++i)
+			{
+                data_size += str_size * col_num[i] + 1;
+                res_offsets[i] = data_size;
+            }
+            res_data.resize(data_size);
+            for (UInt64 i = 0; i < col_num.size(); ++i)
+			{
+                array((UInt8 *)copy_str.data(), res_data.data() + res_offsets[i - 1], str_size + 1, col_num[i]);
+            }
+        }
+
+    private:
+        template <typename T>
+        static void array(const UInt8 * src, UInt8 * dst, const UInt64 size, T repeatTime)
+        {
+            UInt64 i = 0;
+            do
+            {
+                memcpy(dst, src, size - 1);
+                dst += size - 1;
+                ++i;
+            } while (i < repeatTime);
+            *dst = 0;
+        }
+    };
+
+
+    template <typename Impl>
+    class FunctionRepeatImpl : public IFunction
+        {
+            template <typename F>
+            static bool castType(const IDataType * type, F && f)
+            {
+                return castTypeToEither<
+                    DataTypeUInt8,
+                    DataTypeUInt16,
+                    DataTypeUInt32,
+                    DataTypeUInt64>(type, std::forward<F>(f));
+            }
+
+        public:
+			static constexpr auto name = "repeat";
+            static FunctionPtr create(const Context &) { return std::make_shared<FunctionRepeatImpl>(); }
+
+            String getName() const override { return name; }
+
+			size_t getNumberOfArguments() const override { return 2; }
+
+			DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+			{
+                if (!isString(arguments[0]))
+                    throw Exception(
+                        "Illegal type " + arguments[0]->getName() + " of argument of function " + getName(),
+                        ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+                if (!isUnsignedInteger(arguments[1]))
+                    throw Exception(
+                        "Illegal type " + arguments[1]->getName() + " of argument of function 1" + getName(),
```

Comment:
It will look like `of argument of function 1repeat`

Reply:
Sorry, it added during debugging, and forgot to delete.